pkg/cmd/server/origin/master_config.go

```diff
@@ -196,7 +201,7 @@ func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
 	}
 	admissionController := admission.NewChainHandler(plugins...)
 
-	serviceAccountTokenGetter, err := newServiceAccountTokenGetter(options, client)
+	serviceAccountTokenGetter, err := newServiceAccountTokenGetter(options, kubeEtcdClient)
```

Comment:
you can stop passing the client in at all and have `newServiceAccountTokenGetter` build the etcd client if `KubernetesMasterConfig` is presetn